builtin/providers/mssql/resource_database.go

```diff
@@ -0,0 +1,103 @@
+package mssql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceMSsqlDatabase() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSsqlDatabaseCreate,
+		Read:   resourceMSsqlDatabaseRead,
+		Delete: resourceMSsqlDatabaseDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceMSsqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Client)
+	conn, err := client.Connect()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	dbName := d.Get("name").(string)
+
+	_, err = conn.Query("CREATE DATABASE [" + dbName + "]")
```

Comment:
Probably want to use the [parameters support](https://github.com/denisenkom/go-mssqldb#statement-parameters) rather than string concatenation to ensure that everything is quoted properly and avoid a visit from [bobby drop tables](https://xkcd.com/327/). :grinning: